### PR TITLE
TagV2

### PR DESCRIPTION
**Pull request by Crenshaw1312**
So I don't really know how to make a draft of one of these I couldn't get the branches to compare and stuff...
I plan on adding more syntax as I see fit, read the comment at the top for the changes I have made. Indigo is my staple, please leave it as indigo.
For a demo join my smol guild: https://discord.gg/y3BH7rr74w 
I think this is better fit as TagV2 since I added the syntax to it and a bunch of other things, I'll make a PR later to improve some basic things on the original tag system.

	Changes: 
	 Syntax
	 better color (indigo)
	 new `;tag list` appearence
	 RoleFilter on tag management
	 Response is now an embed
	 image support
	 Each sub command has more options
	 Shows who it was requested by

# Syntax
I plan on adding Args support possibly idk. give a thumbs up if you like the idea.
## Current
**USER**
{{User-String}}: User's name Crenshaw#1312
{{User-Username}}: User's username Crenshaw
{{User-Discrminator}}: 4 digits after username 1312
{{User-ID}}: User's ID 766385575530856458
{{User-Age}}: User's age 2 weeks 4 days and 1 hour
{{User-Mention}}:Mentions triggering user @Crenshaw
**CHANNEL**
{{Channel-Name}}: Channel name of channel pagstdb
{{Channel-Topic}}: Channel topic some topic for pagstdb
{{Channel-ID}}: Channel ID of trggering channel 769320349078650932

## Example
Screen_Shot_2020-11-02_at_10 34 10_AM

# Tag List
It now uses the css code block and is spaced out so tags don't get jumbled together and become a mess, I also had to remove certain characters do to splitting, and coloration

For now, or ever, there is no pagination via reactions. I am not a fan of it but I may add it so others can use it if the enjoy it, I'll try and get it to work with Joe's current one so a new pagination doesn't have to made.
Screen Shot 2020-11-02 at 1 36 02 PM

### CODE REVIEW

File: tag/TagV@.cc.lua
Comment (Crenshaw7860) [inline tag/TagV@.cc.lua:192]: I made the list a code block with css, so the name is blue, aliases yellow, and date is in red. I'm having trouble getting rid of the tag name among the aliases, although I may leave it so that if a tag has no name that spot is still filled.

**Review by Crenshaw7860 — COMMENTED** [review]
My first review to organize myself a bit, also so y'all know what I did and where things are.

**Comment by Crenshaw7860** [comment]
I'm really bad at this whole pull request thing, this is a mess

**Comment by thewilloftheshadow** [comment]
*glances at the mess that's known as #90*

File: tag/TagV@.cc.lua
Comment (thewilloftheshadow) [reply tag/TagV@.cc.lua]: Blurple is superior

**Comment by Crenshaw1312** [comment]
I AM DONE
@Jo3-L review and sumbit pls, leave color as indigo

**Comment by Crenshaw1312** [comment]
could I delete `;tag info`? since the `;tag list` command essentially does it

**Review by Jo3-L — CHANGES_REQUESTED** [review]
I've discussed some things I would like to see in this PR before merging on Discord w/ @Crenshaw1312. Please request a review when you think you're finished, and feel free to contact me here or over on Discord if you have further questions. Converting to draft PR for now.

**Comment by Jo3-L** [comment]
Oh, just a comment, would like this "syntax" to be compatible (i.e be a subset of) YAGPDB CC if it is to be kept. I am skeptical as to the benefits as, well, users should use YAGPDB CC directly if they require actual scripting other than embeds + content, but up to you. Example:

```golang
{{ $data := sdict "User" (sdict "Username"  .User.Username) }}
{{ range reFindAllSubmatches `{{\s*((?:\.\w+)+)\s*}}` $USER_INPUT }}
    {{- $path := split (slice (index . 1) 1) "." }}
    {{- $val := $data }}
    {{- range $key := $path }}
        {{- /* can't use `and` because it doesn't short circuit */}}
        {{- if ne (kindOf $val) "map" }}
            {{- with $val.Get $key }}
                {{- $val = . }}
            {{- else }}
                {{- $val = "" }} {{- /* or another appropriate default value */}}
            {{- end }}
        {{- end -}}
    {{- end }}
    {{- /* prevent making too many reReplace call by split + join */}}
    {{- $USER_INPUT = joinStr $val (split $USER_INPUT (index . 0)) -}}
{{ end }}
```

Haven't tested this but that's the general idea.

I find parsing this generally a pain (other than the lazy approach demonstrated above) so that's why I'm not a huge fan of including this, again, your call to make though.

**Comment by Jo3-L** [comment]
Please use the `.go.tmpl` file extension as of latest commit. Thanks!

**Comment by Jo3-L** [comment]
Hi @Crenshaw1312, are you still interested in working on this?

**Comment by Aldiwildan77** [comment]
interesting should be merged soon.

**Comment by jo3-l2** [comment]
Stale, no activity for 2 months. Closing for now.
If you'd like to continue working on this ping me @Crenshaw1312.

**Pull request closed without merging**
